replication.go

```diff
@@ -0,0 +1,117 @@
+	ValidateNewReplication(context.Context, CreateReplicationRequest) error
+
+	// GetReplication returns metadata about the replication with the given ID.
+	GetReplication(context.Context, platform.ID) (*Replication, error)
+
+	// UpdateReplication updates the settings for the replication with the given ID.
+	UpdateReplication(context.Context, platform.ID, UpdateReplicationRequest) (*Replication, error)
+
+	// ValidateUpdatedReplication valdiates that a replication is still usable after applying the
+	// given update, without persisting the new configuration.
+	ValidateUpdatedReplication(context.Context, platform.ID, UpdateReplicationRequest) error
+
+	// DeleteReplication deletes all info for the replication with the given ID.
+	DeleteReplication(context.Context, platform.ID) error
+
+	// ValidateReplication checks that the replication with the given ID is still usable with its
+	// persisted settings.
+	ValidateReplication(context.Context, platform.ID) error
+}
+
+// Replication contains all info about a replication that should be returned to users.
+type Replication struct {
+	ID                    platform.ID `json:"id"`
+	OrgID                 platform.ID `json:"orgID"`
+	Name                  string      `json:"name"`
+	Description           *string     `json:"description,omitempty"`
+	RemoteID              platform.ID `json:"remoteID"`
+	LocalBucketID         platform.ID `json:"localBucketID"`
+	RemoteBucketID        platform.ID `json:"remoteBucketID"`
+	MaxQueueSizeBytes     int64       `json:"maxQueueSizeBytes"`
+	CurrentQueueSizeBytes int64       `json:"currentQueueSizeBytes"`
+	LatestResponseCode    *int32      `json:"latestResponseCode,omitempty"`
+	LatestErrorMessage    *string     `json:"latestErrorMessage,omitempty"`
+}
+
+// ReplicationListFilter is a selection filter for listing replications.
+type ReplicationListFilter struct {
+	OrgID         platform.ID
+	Name          *string
+	RemoteID      *platform.ID
+	LocalBucketID *platform.ID
+}
+
+// Replications is a collection of metadata about replications.
+type Replications struct {
+	Replications []Replication `json:"replications"`
+}
+
+// CreateReplicationRequest contains all info needed to establish a new replication
+// to a remote InfluxDB bucket.
+type CreateReplicationRequest struct {
+	OrgID             platform.ID `json:"orgID"`
+	Name              string      `json:"name"`
+	Description       *string     `json:"description,omitempty"`
+	RemoteID          platform.ID `json:"remoteID"`
+	LocalBucketID     platform.ID `json:"localBucketID"`
+	RemoteBucketID    platform.ID `json:"remoteBucketID"`
+	MaxQueueSizeBytes int64       `json:"maxQueueSizeBytes,omitempty"`
+}
+
+func (r *CreateReplicationRequest) OK() error {
+	if r.MaxQueueSizeBytes < MinReplicationMaxQueueSizeBytes {
+		return &ErrMaxQueueSizeTooSmall
+	}
+
+	return nil
+}
+
+// UpdateReplicationRequest contains a partial update to existing info about a replication.
+type UpdateReplicationRequest struct {
```

Comment:
What would this do if none of the parameters were provided?

Reply:
It could be a validation error, or it could be a no-op that returns the current state of the replication. Not opinionated either way (my thought for the `remotes` side of the implementation was to make it a no-op, but in the code I wrote it looks like it might cause an ISE as-is).

Reply:
Hmmm actually, the test case I just wrote shows the no-op case works for the existing remotes service. Will keep & replicate that behavior for replications unless you think it's more confusing than useful.

Reply:
Seems fine to keep it as-is for now...doesn't seem like it is very likely that clients would send an empty update request, but if they do it seems reasonable that they would get the result back un-changed.